app/models/annotation_slicer.rb

```diff
@@ -0,0 +1,49 @@
+class AnnotationSlicer
+  def initialize(original_text, original_denotations, original_relations)
+    @text = original_text
+    @denotations = original_denotations || []
+    @relations = original_relations || []
+  end
+
+  def annotation_in(range)
+    denotations = denotations_in range
+    relations = relations_of denotations
+    chunk_text = @text[range.begin...range.end]
+    {
+      "text" => chunk_text,
+      "denotations" => denotations,
+      "relations" => relations
+    }
+  end
+
+  def denotations_in(range)
+    @denotations.map do |d|
+      d_start = d["span"]["begin"]
+      d_end = d["span"]["end"]
+      if range.begin <= d_start && d_end <= range.end
```

Comment:
```suggestion
      if range.cover?(d_start..d_end)
```
て書けるっぽいです。
僕は使ったことがないので、動作に確信はありません。

https://docs.ruby-lang.org/ja/latest/method/Range/i/cover=3f.html

Reply:
https://github.com/jdkim/textae_campus/pull/15/commits/a081ebd89ecd17367b14bc9b8d1f5e7ce4a407a1 で使ってみました。テストは通ります。